dotnet/src/SemanticKernel.Abstractions/Kernel.cs

```diff
@@ -88,6 +93,12 @@ public sealed class Kernel
         {
             this._promptFilters = new(promptFilters);
         }
+
+        // Copy the original options to prevent them from being mutated.
```

Comment:
@eiriktsarpalis thanks for helping to identify the right solution for JSO in SK. Please review it when you have time.